internal/providers/vault/provider.go

```diff
@@ -43,6 +46,20 @@ func (p *Provider) GetName() string {
 // hash returned by the Vault.
 const VaultDefaultField = "value"
 
+// VaultBackendKVv2 is the Vault backend KV version 2.
+const VaultBackendKVv2 = "kv-v2"
+
+// parseVaultBackend returns the (optional) backend of the backend from the id.
+func parseVaultBackend(id string) (string, string) {
+	tokens := strings.SplitN(id, ":", 2)
+	switch len(tokens) {
+	case 1:
+		return "", id
+	default:
+		return tokens[0], tokens[1]
+	}
+}
```

Comment:
We should raise an explicit error here if the backend ID is not supported

Reply:
This ties into my other comment about `kv:` prefix but at the very least yes we should trow an error if a backend is provided and we don't support it.

Reply:
Given new design, this should no longer be relevant.